src/arim/geometry.py

```diff
@@ -1457,11 +1459,144 @@ def points_in_rectbox(
 )
 
 
-def points_1d_wall_z(xmin, xmax, z, numpoints, y=0.0, name=None, dtype=None):
+def make_contiguous_geometry(coords, numpoints, names=None, dtype=None):
+    """
+    Returns a list of OrientedPoints with length m which are uniquely named.
+    Default naming convention defines the frontwall as the wall drawn between
+    the first pair of points iff z=0.0; backwalls have constant z; sidewalls 
+    have constant x; and otherwalls are anything else.
+
+    Parameters
+    ----------
+    coords : ndarray[float]
+        Walls drawn between each point in coords. Shape must be (m+1, 2) or
+        (m+1, 3).
+    numpoints : int or ndarray[int]
+        Number of points which each wall is split into. If ndarray, shape must
+        be (m,) a list of values for each wall individually.
+    names : list[str] or None, optional
+        List of names for each wall, which must be unique. If None, the default
+        naming convention will be used.
+    dtype : numpy.dtype, optional
+
+    Returns
+    -------
+    list[OrientedPoints].
+
+    """
+    if dtype is None:
+        dtype = s.FLOAT
+    
+    coords = np.squeeze(coords)
+    if coords.shape[0] < 2:
+        raise ValueError(
+            "Not enough coordinates provided to draw lines for geometry."
+        )
+    if coords.shape[1] not in [2, 3]:
+        raise ValueError("Coordinates should be 2D or 3D.")
+    
+    numpoints = np.squeeze(numpoints).ravel().astype(int)
+    if numpoints.shape[0] == 1:
+        numpoints = numpoints[0] * np.ones(coords.shape[0] - 1, dtype=int)
+    else:
+        if numpoints.shape[0] != coords.shape[0] - 1:
+            raise ValueError("Too many / few values of `numpoints` provided.")
+        
+    if names is not None:
+        if len(names) != coords.shape[0] - 1:
+            raise ValueError("Too many / few wall names provided.")
+    else:
+        bw_idx, sw_idx, ow_idx = 0, 0, 0
+    
+    walls = []
+    for idx, (start, end) in enumerate(zip(coords[:-1, :], coords[1:, :])):
+        if numpoints.shape[0] == 1:
+            n = numpoints[0]
+        else:
+            n = numpoints[idx]
+        
+        if names is None:
```

Comment:
I'm not fully convinced by this nomenclature, because it seems not very generalizable: the block may have multiple layers (but certainly only one backwall), the side walls could be angled, etc.

How about forcing the caller to provide explicit names if desired, and to use "wall_0", "wall_1", etc as a default otherwise?

Reply:
Good point, I'd left this in directly for my use but it would be better to have a more general naming convention and let the user choose their own names if not suitable.